lib/src/config.rs

```diff
@@ -0,0 +1,207 @@
+//! Repository config and metadata
+
+use std::{io, fs};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+use sodiumoxide::crypto::{pwhash, secretbox, box_};
+use serialize::hex::{ToHex};
+
+use {base64, serde_yaml};
+use serde::{self, Deserialize};
+
+pub const REPO_VERSION_LOWEST: u32 = 0;
+pub const REPO_VERSION_CURRENT: u32 = 1;
+
+pub const DATA_SUBDIR: &'static str = "chunk";
+pub const NAME_SUBDIR: &'static str = "name";
+pub const INDEX_SUBDIR: &'static str = "index";
+
+pub const LOCK_FILE: &'static str = ".lock";
+pub const PUBKEY_FILE: &'static str = "pub_key";
+pub const SECKEY_FILE: &'static str = "sec_key";
+pub const VERSION_FILE: &'static str = "version";
+pub const CONFIG_YML_FILE: &'static str = "config.yml";
+
+pub fn lock_file_path(path: &Path) -> PathBuf {
+    path.join(LOCK_FILE)
+}
+
+pub fn pub_key_file_path(path: &Path) -> PathBuf {
+    path.join(PUBKEY_FILE)
+}
+
+pub fn sec_key_file_path(path: &Path) -> PathBuf {
+    path.join(SECKEY_FILE)
+}
+
+pub fn version_file_path(path: &Path) -> PathBuf {
+    path.join(VERSION_FILE)
+}
+
+pub fn config_yml_file_path(path: &Path) -> PathBuf {
+    path.join(CONFIG_YML_FILE)
+}
+
+pub fn write_seckey_file(path: &Path,
+                     encrypted_seckey: &[u8],
+                     nonce: &secretbox::Nonce,
+                     salt: &pwhash::Salt)
+                     -> io::Result<()> {
+    let mut seckey_file = fs::File::create(path)?;
+    let mut writer = &mut seckey_file as &mut Write;
+    writer.write_all(&encrypted_seckey.to_hex().as_bytes())?;
+    writer.write_all(&nonce.0.to_hex().as_bytes())?;
+    writer.write_all(&salt.0.to_hex().as_bytes())?;
+    writer.flush()?;
+    Ok(())
+}
+
+pub fn write_version_file(repo_path : &Path, version : u32) -> super::Result<()> {
+    let path = version_file_path(&repo_path);
+    let path_tmp = path.with_extension("tmp");
+    let mut file = try!(fs::File::create(&path_tmp));
+    {
+        let mut writer = &mut file as &mut Write;
+        write!(writer, "{}", version)?;
+    }
+
+    file.flush()?;
+    file.sync_data()?;
+
+    fs::rename(path_tmp, &path)?;
+
+    Ok(())
+}
+
+pub fn write_config_v0(repo_path: &Path, pk: box_::PublicKey, sk: box_::SecretKey, passphrase : &str) -> super::Result<()> {
+    {
+        let pubkey_path = pub_key_file_path(&repo_path);
+
+        let mut pubkey_file = try!(fs::File::create(pubkey_path));
```

Comment:
You are absolutely right. https://github.com/japaric/untry Though I'll just run `untry` on the whole repo after this PR gets merged. Otherwise it's just a lot of noise, unrelated to the core change.